spec/components/product_component_spec.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe ProductComponent, type: :component do
+  let(:product) { create(:simple_product) }
+
+  describe 'unit' do
+    before do
+      render_inline(
+        ProductComponent.new(
+          product: product, columns: [{ label: "Unit", value: "unit", sortable: false }]
+        )
+      )
+    end
+
+    it 'concatenates the unit value and the unit description' do
+      expect(page.find('.unit')).to have_content '1.0 weight'
+    end
+  end
+
+  describe 'category' do
+    let(:product) do
+      product = create(:simple_product)
+      product.taxons << create(:taxon, name: 'random')
+
+      product
+    end
+
+    before do
+      render_inline(
+        ProductComponent.new(
+          product: product, columns: [{ label: "Category", value: "category", sortable: false }]
+        )
+      )
+    end
+
+    it "joins the categories' name" do
+      expect(page.find('.category')).to have_content(product.taxons.map(&:name).join(', '))
+    end
+  end
+
+  describe 'on_hand' do
+    let(:product) { create(:simple_product, on_hand: on_hand) }
+    let(:on_hand) { 5 }
+
+    before do
+      render_inline(
+        ProductComponent.new(
+          product: product, columns: [{ label: "On Hand", value: "on_hand", sortable: false }]
+        )
+      )
+    end
+
+    it 'return product on_hand' do
```

Comment:
It's minor, but it's helpful to have consistency of language used, even in labels such as these. The correct tense would be "returns", same as below.
```suggestion
    it 'returns product on_hand' do
```